forest_benchmarking/benchmarks/classical_logic_circuits/classical_reversible_logic_primitives.py

```diff
@@ -0,0 +1,326 @@
+"""
+Circuit primitives for classical reversible logic
+
+At the moment it is primarily using the simple adder construction in
+
+[CDKM96] A new quantum ripple-carry addition circuit
+         Cuccaro, Draper, Kutin, and Moulton
+         https://arxiv.org/abs/quant-ph/0410184
+
+There are many other classical logic primitives that can be coded see
+e.g.
+
+[VBE96] Quantum networks for elementary arithmetic operations
+        Vedral,  Barenco, Ekert
+        Phys. Rev. A 54, 147 (1996)
+        https://doi.org/10.1103/PhysRevA.54.147
+        https://arxiv.org/abs/quant-ph/9511018
+"""
+import numpy as np
+
+from pyquil.quil import Program
+from pyquil.gates import CNOT, CCNOT, X, I, H, CZ
+
+
+def majority_gate(a, b, c, CNOTfun = CNOT, CCNOTfun = CCNOT):
```

Comment:
try to get as close as possible to pep8 variable/function names. It's not straightforward since `CNOT` and `CCNOT` don't follow pep8 for function names. I suggest `CNOT_func` and `CCNOT_func`.